Bottom: 7d20867fd239ebf6366b5c1b3d296d69b8a7fcb5
Top:    91a2d3a738624a996a0b5d8e3d6a88deeb8eca00
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-18 16:53:44 +0000

polygon1.c: Refactor poly_Copy0 to use poly_Create(), rather than open-coding

This ensures all polygon initialisation happens in a common place


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 511bd21..0000871 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2815,12 +2815,10 @@ poly_Copy0 (POLYAREA ** dst, const POLYAREA * src)
   if (src == NULL)
     return TRUE;
 
-  *dst = (POLYAREA *)calloc (1, sizeof (POLYAREA));
-  if (*dst == NULL)
+  if ((*dst = poly_Create ()) == NULL || !poly_Copy1 (*dst, src))
     return FALSE;
-  (*dst)->contour_tree = r_create_tree (NULL, 0, 0);
 
-  return poly_Copy1 (*dst, src);
+  return TRUE;
 }
 
 void
